src/templates/prepared-map/usecase-detail/example/structure-tree/parse.ts

```diff
@@ -0,0 +1,205 @@
+import type {
+  EnumDefinitionNode,
+  ListDefinitionNode,
+  NamedFieldDefinitionNode,
+  NamedModelDefinitionNode,
+  ObjectDefinitionNode,
+  PrimitiveTypeNameNode,
+  ProfileDocumentNode,
+  Type,
+  UnionDefinitionNode,
+} from '@superfaceai/ast';
+
+import type {
+  ExampleArray,
+  ExampleObject,
+  ExampleScalar,
+  UseCaseExample,
+} from '../usecase-example';
+
+export function parse(ast: ProfileDocumentNode, type: Type): UseCaseExample {
+  const namedModelDefinitionsCache: {
+    [key: string]: NamedModelDefinitionNode;
+  } = {};
+
+  const namedFieldDefinitionsCache: {
+    [key: string]: NamedFieldDefinitionNode;
+  } = {};
+
+  ast.definitions.forEach(definition => {
+    if (definition.kind === 'NamedFieldDefinition') {
+      namedFieldDefinitionsCache[definition.fieldName] = definition;
+    } else if (definition.kind === 'NamedModelDefinition') {
+      namedModelDefinitionsCache[definition.modelName] = definition;
+    }
+  });
+
+  return visit(type, namedModelDefinitionsCache, namedFieldDefinitionsCache);
+}
+
+export function visit(
+  node: Type,
+  namedModelDefinitionsCache: {
+    [key: string]: NamedModelDefinitionNode;
+  },
+  namedFieldDefinitionsCache: {
+    [key: string]: NamedFieldDefinitionNode;
+  }
+): ExampleArray | ExampleObject | ExampleScalar {
+  switch (node.kind) {
+    case 'ObjectDefinition':
+      return visitObjecDefinition(
+        node,
+        namedModelDefinitionsCache,
+        namedFieldDefinitionsCache
+      );
+    case 'PrimitiveTypeName':
+      return visitPrimitiveNode(node);
+    case 'ListDefinition':
+      return visitListNode(
+        node,
+        namedModelDefinitionsCache,
+        namedFieldDefinitionsCache
+      );
+    case 'EnumDefinition':
+      return visitEnumNode(node);
+    case 'ModelTypeName': {
+      const foundNode = namedModelDefinitionsCache[node.name];
+      if (foundNode.type === undefined) {
+        throw new Error('Type not found');
+      }
+
+      return visit(
+        foundNode.type,
+        namedModelDefinitionsCache,
+        namedFieldDefinitionsCache
+      );
+    }
+    case 'NonNullDefinition':
+      return visit(
+        node.type,
+        namedModelDefinitionsCache,
+        namedFieldDefinitionsCache
+      );
+    case 'UnionDefinition':
+      return visitUnionNode(
+        node,
+        namedModelDefinitionsCache,
+        namedFieldDefinitionsCache
+      );
+    default:
+      throw new Error(`Invalid kind: ${node?.kind ?? 'undefined'}`);
```

Comment:
I'm thinking whether it wouldn't be better to [use exhaustive switch](https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript) here?

It will still throw in runtime but should be a better guard type-wise.